runtime/arc.js

```diff
@@ -174,7 +174,8 @@ export class Arc {
             context.dataResources.set(storageKey, storeId);
             // TODO: can't just reach into the store for the backing Store like this, should be an 
             // accessor that loads-on-demand in the storage objects.
-            await this._serializeHandle(handle._backingStore, context, storeId);
+            await handle.ensureBackingStore();
```

Comment:
Here and the changes below in this file, are there unit tests that would be useful to add to cover these changes?

I'd infer that we had to make these changes because other work in this PR caused existing tests to fail. So in that sense they are covered, but in another sense they were broken previously and we didn't know, so maybe more granular tests for the specific cases would be helpful.

I am ok if you think it's fine as is, wanted to ask while reading.

Reply:
Yup, these changes are related to existing test failures. Most of these tests started failing because of the situation I described in the headline comment - duplicated handles leading to a lack of coverage of concurrency mechanisms. In that sense, they were broken previously because of an unrelated component's failure. This is obviously concerning in other ways - how many other tests don't currently test what they are supposed to be testing?

I would like to add a bunch more tests around the channel, but I need to work out the right way to do this and I think that the coverage for this patch is probably sufficient.